lgc/include/lgc/state/PipelineState.h

```diff
@@ -296,6 +296,12 @@ class PipelineState final : public Pipeline {
   // Set error message to be returned to the client by it calling getLastError
   void setError(const llvm::Twine &message);
 
+  // Set the packable state of generic input/output
+  void setInOutPackState();
+
+  // Get whehter the specified input/output of the shader stage can be packed
+  bool canPack(ShaderStage shaderStage, unsigned idx) const { return m_inOutPackStates[shaderStage][idx]; }
```

Comment:
I think we can change idx to a boolean flag like this:
`unsigned idx -> bool output`
`return m_inOutPackStates[shaderStage][idx] -> return m_inOutPackStates[shaderStage][output ? 1 : 0];`

Reply:
Or even better, an enum.

Reply:
Done. Use an enum.